curriculum/challenges/english/blocks/lecture-understanding-object-oriented-programming-and-encapsulation/68c128cbd77e4ba9ed671937.md

```diff
@@ -0,0 +1,260 @@
+---
+id: 68c128cbd77e4ba9ed671937
+title: What are Getters and Setters
+challengeType: 19
+dashedName: what-are-getters-and-setters
+---
+
+# --description--
+
+Getters and setters are methods that let you control how the attributes of a class are accessed and modified. With getters you retrieve a value, and with setters you set a value.
+
+These actions are done through what’s known as properties. They are what what connect getters and setters, and allow access to data.
+
+Properties act like attributes but behave like methods under the hood. Think of them as data you define like methods, but work like attributes. This means you can access properties with dot notation instead of parentheses or round brackets.
+
+The main thing properties do is that they run extra logic behind the scenes when you get, set, or delete values with them. This makes them the perfect choice when you want to access or manipulate data within objects.
+
+So why use properties for that instead of methods? It’s mostly about readability and convention. They make your code cleaner and easier to read.
```

Comment:
```suggestion
So why use properties for that instead of methods? It's mostly about readability and convention. They make your code cleaner and easier to read.
```